### PR TITLE
Add support for VLLM engine V1 in Qwen2.5-VL.

### PR DESCRIPTION
**Pull request by viktorpusTT**
### Ticket
https://github.com/tenstorrent/vllm/issues/231

Related PR to VLLM: https://github.com/tenstorrent/vllm/pull/257

CI job: https://github.com/tenstorrent/tt-metal/actions/runs/19859950135

### Problem description
Add support to vLLM engine V1 for this model.

### What's changed
* Switch to standard Qwen25 MultiModalProcessor and DummyInputsBuilder instead of TT-specific.
* Reconstruct attention mask from other inputs for V1.
* Some code branching for V0 vs V1 was needed.

### Checklist
- [ ] [All post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/all-post-commit-workflows.yaml) CI passes
- [ ] [Blackhole Post commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml) CI with demo tests passes (if applicable)
- [ ] [Model regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml) CI passes (if applicable)
- [ ] [Device performance regression](https://github.com/tenstorrent/tt-metal/actions/workflows/perf-device-models.yaml) CI passes (if applicable)
- [ ] (For models and ops writers) [Single-card demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/single-card-demo-tests.yaml) CI passes (if applicable) See [recommended dev flow](https://github.com/tenstorrent/tt-metal/blob/main/models/docs/MODEL_ADD.md#a-recommended-dev-flow-on-github-for-adding-new-models).
- [ ] [Galaxy quick](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-quick.yaml) CI passes (if applicable)
- [ ] [Galaxy demo tests, for Llama](https://github.com/tenstorrent/tt-metal/actions/workflows/galaxy-demo-tests.yaml) CI passes, if applicable, because of current Llama work
- [ ] (For runtime and ops writers) [T3000 unit tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-unit-tests.yaml) CI passes (if applicable, since this is run on push to main)
- [ ] (For models and ops writers) [T3000 demo tests](https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-demo-tests.yaml) CI passes (if applicable, since this is required for release)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: models/demos/qwen25_vl/tt/generator_vllm.py [hunk 1]
```diff
@@ -179,28 +186,79 @@ def prefill_forward(
...
+                    else torch.ones_like(tokens[i : i + 1], dtype=tokens.dtype)
+                    for i, im in enumerate(kwargs["images"])
+                ],
+                dim=0,
+            )
 
         # Prepare text + vision inputs for decoder model
         text_embeds = self.reference_model.model.language_model.embed_tokens(inputs.input_ids)
+
+        if envs.VLLM_USE_V1:
```
Comment (skhorasganiTT) [inline models/demos/qwen25_vl/tt/generator_vllm.py]: Can this v1/v0 if-else be merged with the one above? device can be obtained without text_embeds.

**Review by gwangTT — COMMENTED** [review]
LGTM. Thanks, Viktor!

**Review by gwangTT — APPROVED** [review]

**Review by skhorasganiTT — APPROVED** [review]
